lib/standalone.mjs

```diff
@@ -0,0 +1,323 @@
+			return absoluteFilepath;
+		});
+
+		const getStylelintResults = absoluteFilePaths.map(async (absoluteFilepath) => {
+			debug(`Processing ${absoluteFilepath}`);
+
+			try {
+				const postcssResult = await lintSource(stylelint, {
+					filePath: absoluteFilepath,
+					cache: useCache,
+				});
+
+				if (
+					(postcssResult.stylelint.stylelintError || postcssResult.stylelint.stylelintWarning) &&
+					useCache
+				) {
+					debug(`${absoluteFilepath} contains linting errors and will not be cached.`);
+					stylelint._fileCache.removeEntry(absoluteFilepath);
+				}
+
+				/**
+				 * If we're fixing, save the file with changed code
+				 */
+				if (
+					postcssResult.root &&
+					postcssResult.opts &&
+					!postcssResult.stylelint.ignored &&
+					fix &&
+					!postcssResult.stylelint.disableWritingFix
+				) {
+					const fixedCss = postcssResult.root.toString(postcssResult.opts.syntax);
+
+					if (
+						postcssResult.root &&
+						postcssResult.root.source &&
+						postcssResult.root.source.input.css !== fixedCss
+					) {
+						await writeFileAtomic(absoluteFilepath, fixedCss);
+					}
+				}
+
+				return createPartialStylelintResult(postcssResult);
+			} catch (error) {
+				// On any error, we should not cache the lint result
+				stylelint._fileCache.removeEntry(absoluteFilepath);
+
+				return handleError(error);
+			}
+		});
+
+		stylelintResults = await Promise.all(getStylelintResults);
+	} else if (allowEmptyInput ?? config?.allowEmptyInput ?? (await canAllowEmptyInput(stylelint))) {
+		stylelintResults = await Promise.all([]);
+	} else if (filePathsLengthBeforeIgnore) {
+		// All input files ignored
+		stylelintResults = await Promise.reject(new AllFilesIgnoredError());
+	} else {
+		stylelintResults = await Promise.reject(new NoFilesFoundError(fileList));
+	}
+
+	if (useCache) {
+		stylelint._fileCache.reconcile();
+	}
+
+	const result = prepareReturnValue(stylelintResults, maxWarnings, formatterFunction, cwd);
+
+	debug(`Linting complete in ${Date.now() - startTime}ms`);
+
+	return result;
+}
+
+/**
+ * @param {FormatterType | Formatter | undefined} selected
+ * @returns {Formatter}
+ */
+function getFormatterFunction(selected) {
+	if (typeof selected === 'string') {
+		const formatterFunction = formatters[selected];
+
+		if (formatterFunction === undefined) {
+			const formattersText = getFormatterOptionsText(', ', '"');
+
+			throw new Error(`You must use a valid formatter option: ${formattersText} or a function`);
+		}
+
+		return formatterFunction;
+	}
+
+	if (typeof selected === 'function') {
+		return selected;
+	}
+
+	assert(formatters.json);
+
+	return formatters.json;
+}
+
+/**
+ * @typedef {import('stylelint').CssSyntaxError} CssSyntaxError
+ *
+ * @param {unknown} error
+ * @return {import('stylelint').LintResult}
+ */
+function handleError(error) {
+	if (error instanceof Error && error.name === 'CssSyntaxError') {
+		return createPartialStylelintResult(undefined, /** @type {CssSyntaxError} */ (error));
```

Comment:
off topic

> `'createPartialStylelintResult must be called with either postcssResult or CssSyntaxError',`

I think this function should be refactored.
The fact that both arguments are optional is a red flag.
Maybe differentiate based on the content of a single object being passed?
Looks like a code smell to me.

Reply:
I agree that this code needs refactoring, although the function is old. Can you please open an issue or a pull request? I hope it will be after merging `v16` into `main`.